src/DynamoCore/Graph/Nodes/PortModel.cs

```diff
@@ -430,6 +430,16 @@ internal string GetInputPortType()
                 return type;
             }
 
+            var cusNode = Owner as CustomNodes.Function;
+            if (cusNode != null)
```

Comment:
I think @mjkkirschner suggested to use a `is` check on @reddyashish 's PR, you may want to give it a try to simplify the code as well here